Add support for including help files in plugins

danielb987 opened the pull request:
This PR adds support for help files in a plugin. See https://github.com/danielb987/JmriPlugin for an example plugin.

To use this feature, the JMRI web server must be running.

Local help pages are accessed by the path `/help/`, for example:
http://localhost:12080/help/en/html/tools/logixng/LogixNG.shtml

Help pages within a plugin are accessed by the path `/plugin/`, for example:
http://localhost:12080/plugin/se/bergqvist/jmri/LogixNG.shtml

In the plugin https://github.com/danielb987/JmriPlugin, the help pages are in the source package `se.bergqvist.jmri` and the images are in the source package `se.bergqvist.jmri.images`. Help pages in the plugin can link to, and include, JMRI help pages if the path starts with `/help/`.

bobjacobsen reviewed (approved):
Nice solution!

danielb987 commented:
@bobjacobsen 
I have found a problem when running `ant jar` while having my plugin in the `setting/lib` folder:
```
BUILD FAILED
F:\Projekt\Java\GitHub\JMRI\build.xml:1855: No suitable relative path from F:\Projekt\Java\GitHub\JMRI to C:\Users\daniel\JMRI\lib\JmriPlugin.jar
```
Not a big problem since I can delete my plugin jar file, create the jmri.jar file and then copy my plugin back to the settings/lib folder. But it might be worth mention in the documentation for plugins.

danielb987 commented:
@dsand47 
I have updated the Help preferences. Before, the option for the JMRI web server was:
`Open help pages using the web server in JMRI (Experimental. Not recommended)`
I have changed it to:
`Open help pages using the web server in JMRI`

I don't know if there is any help page that should be updated according to this.

dsand47 commented:
> @dsand47 I have updated the Help preferences. Before, the option for the JMRI web server was: `Open help pages using the web server in JMRI (Experimental. Not recommended)` I have changed it to: `Open help pages using the web server in JMRI`
> 
> I don't know if there is any help page that should be updated according to this.

The Preferences help page is not affected.

danielb987 commented:
> > @dsand47 I have updated the Help preferences. Before, the option for the JMRI web server was: `Open help pages using the web server in JMRI (Experimental. Not recommended)` I have changed it to: `Open help pages using the web server in JMRI`
> > I don't know if there is any help page that should be updated according to this.
> 
> The Preferences help page is not affected.

Current master:
![Skärmklipp 2023-12-12 19 17 14](https://github.com/JMRI/JMRI/assets/20255317/847ac4c9-8cae-4fc1-aae9-c8f18887c4ec)

This PR:
![Skärmklipp 2023-12-12 19 24 33](https://github.com/JMRI/JMRI/assets/20255317/1d87719f-8e41-4a6b-b62c-0e56f595e261)

danielb987 commented:
@bobjacobsen 
I think this PRis ready for review and merge. Once this PR is merged and you have updated the documentation for the plugins, it would be good to tell Matt Millican about it so he's aware of the solutions we actually offers now.
